lib/Driver/Compilation.cpp

```diff
@@ -727,6 +731,16 @@ namespace driver {
                  : TaskFinishedResponse::StopExecution;
     }
 
+#if defined(_WIN32)
+    struct FileBinaryModeRAII {
+      FileBinaryModeRAII(FILE *F) : F(F) { _setmode(_fileno(F), _O_BINARY); }
+      ~FileBinaryModeRAII() { _setmode(_fileno(F), _O_TEXT);}
+      FILE *F;
+    };
+#else
+    struct ChangeStderrToBinaryMode {};
```

Comment:
This is defined but not used, is intentional?

Reply:
Looks like this is meant to be:
```c++
struct FileBinaryModeRAII {
  FileBinaryModeRAII(FILE *) {}
};
```
So the code compile as a no-op on non-Windows?

Reply:
Good catch. Done.